File: 0338-counting-bits/0338-counting-bits.java
class Solution {    
    public int[] countBits(int n) {
		int [] output = new int[n+1];
		for(int i=0;i<=n;i++) {
		    if(i==0)
		        output[0] = 0;
		    else if(i==1)
		        output[1] = 1;
		    else {
		        int temp = (i%2==0) ? output[i/2] : output[i/2]+1;
		        output[i] = temp;
		    }
		}
        return output;        
    }
}

File: 0338-counting-bits/0338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        output = []
        for i in range(n+1):
            if i == 0:
                output.append(0)
            elif i == 1:
                output.append(1)
            else:
                res = output[i//2] if i%2==0 else output[i//2]+1
                output.append(res)
        return output